test/jdk/java/lang/ProcessBuilder/Basic.java

```diff
@@ -2135,10 +2135,33 @@ public void doIt(Map<String,String> environ) {
             final int cases = 4;
             for (int i = 0; i < cases; i++) {
                 final int action = i;
-                List<String> childArgs = new ArrayList<String>(javaChildArgs);
+                List<String> childArgs = new ArrayList<>(javaChildArgs);
+                final ProcessBuilder pb = new ProcessBuilder(childArgs);
+                {
+                    // Redirect any child VM error output away from the stream being tested
+                    // and to the log file. For background see:
+                    // 8231297: java/lang/ProcessBuilder/Basic.java test fails intermittently
+                    // Destroying the process may, depending on the timing, cause some output
+                    // from the child VM.
+                    // This test requires the thread reading from the subprocess be blocked
+                    // in the read from the subprocess; there should be no bytes to read.
+                    assert (childArgs.get(1).equals("-XX:+DisplayVMOutputToStderr")) : "Expected arg 1 to be \"-XX:+DisplayVMOutputToStderr\"";
+                    switch (action & 0x1) {
+                        case 0:
+                            childArgs.set(1, "-XX:+DisplayVMOutputToStderr");
+                            pb.redirectError(INHERIT);
+                            break;
+                        case 1:
+                            childArgs.set(1, "-XX:+DisplayVMOutputToStdout");
+                            pb.redirectInput(INHERIT);
```

Comment:
not `redirectOutput()` ?

Reply:
The naming of the methods is from the point of view of the parent, not the child.
So it seems backward, but is correct.

Reply:
Yes, I have to look this up every time. Its not intuitive.